pages/index.tsx

```diff
@@ -1,41 +1,23 @@
-import { useEffect } from "react";
-import { useRouter } from "next/router";
-
-import Loader from "components/templates/Loader/loader";
-import { supabase } from "lib/utils/supabase";
-import useSession from "lib/hooks/useSession";
+import Head from "next/head";
 import { WithPageLayout } from "../interfaces/with-page-layout";
 
-const Home: WithPageLayout = () => {
-  const router = useRouter();
-  const { onboarded } = useSession();
-
-  useEffect(() => {
-    function redirect(destination: string) {
-      router.push(destination);
-    }
-
-    async function getAccess() {
-      const currentUser = await supabase.auth.getSession();
-      if (!currentUser?.data?.session) {
-        redirect("/feed");
-      }
-
-      if (onboarded === undefined) {
-        return;
-      }
-
-      if (currentUser?.data?.session && onboarded === true) {
-        redirect("/hub/insights");
-      } else {
-        redirect("/feed");
-      }
-    }
-
-    getAccess();
-  }, [router, onboarded]);
+export const getServerSideProps = async () => {
```

Comment:
You should never get to the actual page because of middleware, but if for some reason you do, I've added default redirects to `/feed`.